deps/linenoise/linenoise.c

```diff
@@ -1228,3 +1331,101 @@ int linenoiseHistoryLoad(const char *filename) {
     fclose(fp);
     return 0;
 }
+
+int linenoiseReverseSearchModeEnabled(void) {
+    return reverse_search_mode_enabled;
+}
+
+linenoiseHistorySearchResult searchInHistory(char * searchTerm) {
+    linenoiseHistorySearchResult result = {0};
+
+    if (history_len == 0) {
+        return result;
+    }
+
+    int i = cycle_to_next_search_result ? search_result_history_index : history_len-1 ;
+    int original_i = i;
+    
+    if (i <= 0) {
+        i = history_len-1;
+    } else {
+        i--;
+    }
+
+    while (1) {
+        char * found = strstr(history[i], searchTerm);
+        
+        /* 
+            check if we found the same string at another index when cycling, this would be annoying to cycle through
+            as it might appear that cycling isn't working
+        */
```

Comment:
```suggestion
        /* check if we found the same string at another index when cycling, this would be annoying to cycle through
         * as it might appear that cycling isn't working */
```